brave/brave-core
# [Brave News] Adjust subscription metric buckets, fix channel subscriptions

Resolves https://github.com/brave/brave-browser/issues/37845
Resolves https://github.com/brave/brave-browser/issues/37844



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

For the metrics:

0. Start with fresh profile, ensure metrics do not exist.
1. Use News for the first time, ensure that the direct feed and publisher metrics have the value 0, and the channel metric has the value 1.
2. Add/remove some direct feeds, publishers and channels. Ensure that the metrics show the correct values.

For the channel subscription:

Subscribe to the "Celebrities" channel while on the `en-US` locale. Ensure the channel is added to the channel list immediately, and persists after browser restart.
